go-controller/pkg/config/utils.go

```diff
@@ -50,12 +51,20 @@ func ParseClusterSubnetEntries(clusterSubnetCmd string) ([]CIDRNetworkEntry, err
 			return nil, fmt.Errorf("CIDR %q not properly formatted", clusterEntry)
 		}
 
-		var err error
-		_, parsedClusterEntry.CIDR, err = net.ParseCIDR(fmt.Sprintf("%s/%s", splitClusterEntry[0], splitClusterEntry[1]))
+		RawClusterCIDR := fmt.Sprintf("%s/%s", splitClusterEntry[0], splitClusterEntry[1])
+		var (
+			err    error
+			cidrIP net.IP
+		)
+		cidrIP, parsedClusterEntry.CIDR, err = net.ParseCIDR(RawClusterCIDR)
 		if err != nil {
 			return nil, err
 		}
 
+		if bytes.Compare(cidrIP.To16(), parsedClusterEntry.CIDR.IP.To16()) != 0 {
+			return nil, fmt.Errorf("Cluster subnet %s at the middle of the range, correct value should be %s", RawClusterCIDR, parsedClusterEntry.CIDR)
```

Comment:
This will change the current behavior and I would lean towards making it a warning log.
What do you guys think? @jcaamano @dcbw @trozet

Reply:
Yup, I would do a warning.